### ISSUE TITLE
Cannot save the result of get_lines_intensity(plot_result=True)

### ISSUE THREAD

**Issue by jat255** [hyperspy/hyperspy#1884]
Using the latest `RELEASE_next_minor` and the following versions:

```
Python 3.6.4 | packaged by conda-forge
hyperspy (1.4.dev0+git.553.g3ea779ae, c:\users\jat\git_repos\hyperspy)
hyperspy-gui-ipywidgets (1.0.0, c:\users\jat\git_repos\hyperspy_gui_ipywidgets)
hyperspy-gui-traitsui (1.0.3, c:\users\jat\git_repos\hyperspy_gui_traitsui)
pyqt                      4.11.4                   py36_2    conda-forge
matplotlib                2.0.0               np112py36_2    conda-forge
```

The output of `s.get_lines_intensity(plot_result=True)` cannot be saved using the save figure icon in the matplotlib figure. There's no output in the console or in the notebook when clicking the button.

**Comment by jat255** [comment]
Using `plt.gcf().savefig(filename)` does work.

**Comment by jat255** [comment]
This apparently does not have to do with the newlines in the titles, since removing all the `textwrap.fill()` calls in the code does not fix the issue.

**Comment by jat255** [comment]
Sorry for the stream of consciousness... Debugging this more with pyplot, it appears the issue is in fact due to `\n` in the title.

**Comment by jat255** [comment]
`matplotlib` gets the default file name from the window title, so if we want to squash this bug on our end (probably easier than getting something fixed in matplotlib), we should sanitize the window titles.

**Comment by francisco-dlp** [comment]
It is likely that the mpl developers are not aware of this bug, so it may be worth submitting an issue in mpl before going for merging #1885 that, although it fixes the problem on our side, it comes with drawbacks that could be avoided by fixing this in mpl instead. If the behaviour is intended in mpl, we could then merge #1885. @jat255, would you like to open the issue in mpl?

**Comment by jat255** [comment]
@francisco-dlp after a better search, it turns out this is a known issue (https://github.com/matplotlib/matplotlib/issues/9056)

There was a PR proposed, which got discussed, and then rejected and closed without merging - https://github.com/matplotlib/matplotlib/pull/9057. This is why I recommend fixing it on our end, since matplotlib is such a big project that sometimes things don't get included as they should.

If it's preferable, I could make #1885 a little less "aggressive" and just strip out specifically illegal characters (%, :, \n, etc.).

**Comment by francisco-dlp** [comment]
I would suggest to wait a bit to see if it gets properly fixed in mpl.

**Comment by jat255** [comment]
@francisco-dlp It appears there hasn't been any response to the comment on the matplotlib issue.

I think it would be better if we pushed a fix on our side. I've updated #1885 if you care to have a look.

**Comment by ericpre** [comment]
Indeed, it would make sense to have a temporary workaround for this issue in the mean time with a corresponding mention in the code.

**Comment by ericpre** [comment]
Fixed in #1885.
